Source/WebCore/platform/ios/VideoFullscreenInterfaceAVKit.mm

```diff
@@ -98,6 +104,296 @@ @interface AVPlayerViewController (Details)
+@property (retain) AVPlayerController* controller; // Should be called playerController, but can't be due to base class
+@property (retain) AVPlayerLayer* playerLayer;
+@end
+
+//- (AVPlayerController*)playerController
+//{
+//    return _playerController;
+//}
+static AVPlayerController *WebAVPictureInPictureContentViewController_controller(id aSelf, SEL)
+{
+    void* controller;
+    object_getInstanceVariable(aSelf, "_controller", &controller);
+    return static_cast<AVPlayerController*>(controller);
+}
+
+//- (void)setController:(AVPlayerController*)controller
+//{
+//    _playerController = controller;
+//}
+static void WebAVPictureInPictureContentViewController_setController(id aSelf, SEL, AVPlayerController* controller)
+{
+    WebAVPictureInPictureContentViewController *pipController = aSelf;
+    auto* oldController = [pipController controller];
+    if (controller == oldController)
+        return;
+    if (oldController)
+        CFRelease(oldController);
+    if (controller)
+        CFRetain(controller);
+    object_setInstanceVariable(aSelf, "_controller", controller);
+}
+
+//- (AVPlayerLayer*)playerLayer
+//{
+//    return _playerLayer;
+//}
+static AVPlayerLayer *WebAVPictureInPictureContentViewController_playerLayer(id aSelf, SEL)
+{
+    void* layer;
+    object_getInstanceVariable(aSelf, "_playerLayer", &layer);
+    return static_cast<AVPlayerLayer*>(layer);
+}
+
+//- (void)setPlayerLayer:(AVPlayerLayer *)layer
+//{
+//    self.playerLayer = layer;
+//    [self.view.layer addSublayer:layer];
+//}
+static void WebAVPictureInPictureContentViewController_setPlayerLayer(id aSelf, SEL, AVPlayerLayer* layer)
+{
+    WebAVPictureInPictureContentViewController *pipController = aSelf;
+    auto oldPlayerLayer = [pipController playerLayer];
+    if (oldPlayerLayer == layer)
+        return;
+    if (oldPlayerLayer)
+        CFRelease(oldPlayerLayer);
+    if (layer)
+        CFRetain(layer);
+    object_setInstanceVariable(aSelf, "_playerLayer", layer);
+    [[pipController view].layer addSublayer:layer];
+}
+
+//- (void)viewWillLayoutSubviews
+//{
+//    [super viewWillLayoutSubviews];
+//    [_playerLayer setFrame:self.view.bounds];
+//}
+static void WebAVPictureInPictureContentViewController_viewWillLayoutSubviews(id aSelf, SEL)
+{
+    WebAVPictureInPictureContentViewController *pipController = aSelf;
+    objc_super superClass { pipController, getAVPictureInPictureContentViewControllerClass() };
+    auto super_viewWillLayoutSubviews = reinterpret_cast<void(*)(objc_super*, SEL)>(objc_msgSendSuper);
+    super_viewWillLayoutSubviews(&superClass, @selector(viewWillLayoutSubviews));
+    [[pipController playerLayer] setFrame:[pipController view].bounds];
+}
+
+static void WebAVPictureInPictureContentViewController_dealloc(id aSelf, SEL)
+{
+    WebAVPictureInPictureContentViewController *pipController = aSelf;
+    [pipController setValue:nil forKey:@"_controller"];
```

Comment:
Not necessary, but maybe leave a comment saying this is necessary to ensure `CFRelease` is called.